src/jshint.js

```diff
@@ -2218,6 +2217,19 @@ var JSHINT = (function() {
     return this;
   }));
   prefix("new", function() {
+    var mp = metaProperty("target");
+    if (mp) {
+      if (mp.value === "target") {
+        if (!state.inESNext(true)) {
+          warning("W119", state.tokens.prev, "new.target");
+        }
+        if (!state.inClassBody) {
```

Comment:
reading up on this (thanks, I hadn't heard of this before!), I thought it was available in all functions, but not arrow functions ?